src/core/namespace.c

```diff
@@ -1157,6 +1157,15 @@ static int mount_image(const MountEntry *m, const char *root_directory) {
                                 host_os_release_id, host_os_release_version_id, host_os_release_sysext_level);
         if (r == -ENOENT && m->ignore)
                 return 0;
+        if (r == -ESTALE && host_os_release_id)
+                return log_error_errno(r,
```

Comment:
@poettering I know we try to follow some logic with logging vs non-logging functions, but the errors here are really unintelligible. It seems to me it's worth to have this limited exception, so that the actual problem is logged at error level, rather than debug.